src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -406,6 +407,49 @@ public void cookie403Renewal() throws Exception {
     }
 
     /**
+     * This test check that the cookie is renewed if the server presents a Set-Cookie header
+     * after the cookie authentication.
+     * @throws Exception
+     */
+    @Test
+    public void cookieRenewal() throws Exception {
+
+        final String renewalCookieValue =
+                "AuthSession=\"RenewCookie_a2ltc3RlYmVsOjUxMzRBQTUzOtiY2_IDUIdsTJEVNEjObAbyhrgz\";";
+        // Request sequence
+        // _session request to get Cookie
+        // GET request -> 403
+        // _session for new cookie
+        // GET replay -> 200
+        mockWebServer.enqueue(MockWebServerResource.OK_COOKIE);
+        mockWebServer.enqueue(new MockResponse().setResponseCode(200).addHeader("Set-Cookie",
+                String.format(Locale.ENGLISH, "%s;", renewalCookieValue))
+                .setBody("{\"hello\":\"world\"}\r\n"));
+        mockWebServer.enqueue(new MockResponse());
+
+        CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                .username("a")
+                .password("b")
+                .build();
+
+
+        c.executeRequest(Http.GET(c.getBaseUri()));
+        mockWebServer.takeRequest(); // cookie
+        mockWebServer.takeRequest(); // actual get
+
+        //make another get
+        c.executeRequest(Http.GET(c.getBaseUri()));
+
+
+        // also assert that there were 4 calls
+        assertEquals("The server should have received 4 requests", 3, mockWebServer
```

Comment:
Also since `takeRequest` is blocking (potentially until the tests time out if there were not the correct number of requests or an exception thrown) you may want to re-order this so that the assert comes before the `takeRequest` calls.